docs/powerpoint/insert-slides-into-presentation.md

```diff
@@ -0,0 +1,183 @@
+    $("#file").change(storeFileAsBase64);
+    ```
+
+3. Add the following code. About this code, note the following:
+
+    - The `reader.readAsDataURL` method converts the file to base64 and stores it in the `reader.result` property. When the method completes, it triggers the `onload` event handler.
+    - The `onload` event handler trims metadata off of the encoded file and stores the encoded string in a global variable.
+    - The base64-encoded string is stored globally because it will be read by another function that you create in a later step.
+
+    ```javascript
+    let chosenFileBase64;
+
+    async function storeFileAsBase64() {
+        const reader = new FileReader();
+
+        reader.onload = async (event) => {
+            const startIndex = reader.result.toString().indexOf("base64,");
+            const copyBase64 = reader.result.toString().substr(startIndex + 7);
+
+            chosenFileBase64 = copyBase64;
+        };
+
+        const myFile = document.getElementById("file") as HTMLInputElement;
+        reader.readAsDataURL(myFile.files[0]);
+    }
+    ```
+
+## Insert slides with insertSlidesFromBase64
+
+You insert slides from another PowerPoint presentation into the current presentation with the [Presentation.insertSlidesFromBase64](/javascript/api/powerpoint/powerpoint.presentation#insertslidesfrombase64-base64file--options-) method. The following is a simple example in which all of the slides from the source presentation are inserted at the beginning of the current presentation and the inserted slides keep the formatting of the source file. Note that `chosenFileBase64` is a global variable that holds a base64-encoded version of a PowerPoint presentation file.
+
+```javascript
+async function insertAllSlides() {
+  await PowerPoint.run(async function(context) {
+    context.presentation.insertSlidesFromBase64(chosenFileBase64);
+    await context.sync();
+  });
+}
+```
+
+You can control some aspects of the insertion result, including where the slides are inserted and whether they get the source or target formatting (and more), by passing an [InsertSlideOptions](/javascript/api/powerpoint/powerpoint.insertslideoptions) object as a second parameter to `insertSlidesFromBase64`. The following is an example. About this code, note:
+
+- There are two possible values for the `formatting` property: "UseDestinationTheme" and "KeepSourceFormatting". Optionally, you can use the full name of the `InsertSlideFormatting` enum, such as `PowerPoint.InsertSlideFormatting.useDestinationTheme`.
+- The function will insert the slides from the source presentation immediately after the slide specified by the `targetSlideId` property. The value of this property is a string of one of three possible forms: ***nnn*#**, **#*mmmmmmmmm***, or ***nnn*#*mmmmmmmmm***, where *nnn* is the slide's ID (typically 3 digits) and *mmmmmmmmm* is the slide's creation ID (typically 9 digits). Some examples are `267#763315295`, `267#`, and `#763315295`.
```

Comment:
Should we expose this implementation detail? We go into how to programmatically get the slide ID later, which seems like the only real way to get that value. Maybe we should forgo actual values entirely and use a variable like "mySlideId" in this example.

Reply:
I was confused by this, until Onur told me the possible forms. So, I don't think we can do without this. If we postponed this detail, it would only be for maybe one or two steps anyway, so I think its best to just present it as soon as we start talking about the IDs.